bfxapi/websockets/generic_websocket.py

```diff
@@ -146,8 +141,15 @@ def get_authenticated_socket(self):
         self.sockets[sId] = s
         while retries < self.max_retries and self.attempt_retry:
             try:
-                await self._connect(s)
-                retries = 0
+                async with websockets.connect(self.host) as websocket:
+                    self.sockets[sId].set_websocket(websocket)
+                    self.sockets[sId].set_connected()
+                    self.logger.info("Websocket connected to {}".format(self.host))
+                    retries = 0
+                    while True:
+                        await asyncio.sleep(0)
```

Comment:
Do you need the `sleep` call here? Doesn't `websocket.recv` block below?

Reply:
We dont need it, but its a little bit of an optimization hack. It basically forces the the async queue to be cleared before it reads the next incoming message.

Ill add a comment - that probably needs to be more obvious